app/controllers/shelters_controller.rb

```diff
@@ -39,6 +39,7 @@ def update
   def destroy
     shelter = Shelter.find(params[:id])
     if shelter.pets.pending.empty?
+      favorites.delete_pets(shelter.pets.pluck(:id))
```

Comment:
delete all favorited pets from this shelter before deleting it

Reply:
Thought this would be a good way to do it! nice.